clang/lib/Sema/SemaType.cpp

```diff
@@ -6536,6 +6536,30 @@ static void HandleBTFTypeTagAttribute(QualType &Type, const ParsedAttr &Attr,
       ::new (Ctx) BTFTypeTagAttr(Ctx, Attr, BTFTypeTag), Type);
 }
 
+static void handleWrapsAttr(QualType &Type, const ParsedAttr &Attr,
+                            TypeProcessingState &State, bool NoWraps = false) {
+  Sema &S = State.getSema();
+  ASTContext &Ctx = S.Context;
+
+  // wraps and no_wraps are most useful and consistent when used with C. Other
+  // languages have better alternatives within their type systems.
+  if (S.LangOpts.CPlusPlus || S.LangOpts.ObjC)
```

Comment:
This should diagnose.

Reply:
I'm relying on the default diagnostic provided from the LangOpt `COnly` in the table gen. The message is the following:

```
test.cc:2:28: warning: 'wraps' attribute ignored [-Wignored-attributes]
    2 | typedef int __attribute__((wraps)) wrapping_int;
      |                            ^
1 warning generated.
```

Reply:
Ah!  Right... do we need this check then?  Or can this be an assert?

Reply:
For whatever reason, the execution still ends up here during parsing. Can't be an assert :(